### ISSUE TITLE
Zero exit code on secrets-config errors

### ISSUE THREAD

**Issue by farshidtz** [edgexfoundry/edgex-go#4337]
# 🐞 Bug Report
The `secrets-config` command does not consistently exit with the right code. In some error cases, it exits with zero code which complicates the usage by other programs.

### Affected Services [**REQUIRED**]
`secrets-config` CLI app

### Is this a regression?
No

### Description and Minimal Reproduction [**REQUIRED**]
Setup:
```bash
$ snap install edgexfoundry --edge
edgexfoundry (edge) 3.0.0-dev.39 from Canonical✓ installed

$ openssl ecparam -genkey -name prime256v1 -noout -out private.pem && openssl ec -in private.pem -pubout -out public.pem
read EC key
writing EC key

$ KONG_ADMIN_JWT=`sudo cat /var/snap/edgexfoundry/current/secrets/security-proxy-setup/kong-admin-jwt`
```

:heavy_check_mark: Bad algorithm:
```bash
$ edgexfoundry.secrets-config proxy adduser --token-type jwt --user example --algorithm BAD --public_key public.pem --id 1000 --jwt $KONG_ADMIN_JWT
level=ERROR ts=2023-02-08T16:10:46.050346727Z app=secrets-config source=bootstraphandler.go:70 msg="/snap/edgexfoundry/4259/bin/secrets-config proxy adduser: argument --algorithm must be either 'RS256' or 'ES256'"
$ echo $?
2
```

:heavy_check_mark: Missing username:
```bash
$ edgexfoundry.secrets-config proxy adduser --token-type jwt --algorithm ES256 --public_key public.pem --id 1000 --jwt $KONG_ADMIN_JWT
level=ERROR ts=2023-02-08T16:11:02.878690792Z app=secrets-config source=bootstraphandler.go:70 msg="/snap/edgexfoundry/4259/bin/secrets-config proxy adduser: argument --user is required"
$ echo $?
2
```

:x: Bad file input (public key):
```bash
$ edgexfoundry.secrets-config proxy adduser --token-type jwt --user example --algorithm ES256 --public_key bad.pem --id 1000 --jwt $KONG_ADMIN_JWT
level=ERROR ts=2023-02-08T16:10:13.485267104Z app=secrets-config source=bootstraphandler.go:77 msg="failed to read public key from file bad.pem: open bad.pem: no such file or directory"
$ echo $?
0
```

:x: Bad JWT token (Kong Admin token):
```bash
$ edgexfoundry.secrets-config proxy adduser --token-type jwt --user example --algorithm ES256 --public_key public.pem --id 1000 --jwt badtoken
level=ERROR ts=2023-02-08T16:24:58.209231341Z app=secrets-config source=command.go:168 msg="{\"message\":\"Bad token; invalid JSON\"}"
level=ERROR ts=2023-02-08T16:24:58.209262189Z app=secrets-config source=bootstraphandler.go:77 msg="Create consumer request failed with code: 401"
$ echo $?
0
```

## 🌍  Your Environment
**Deployment Environment:**

**EdgeX Version [**REQUIRED**]:** 3.0.0-dev.39

**Anything else relevant?**

As a workaround, the program using `secrets-config` could look for "ERROR" in the standard output instead of checking the exit code.

**Comment by bnevis-i** [comment]
@farshidtz Do any of these fixes need to be backported?  Some of these sub-commands are going away in EdgeX 3.0 (Minnesota)

**Comment by farshidtz** [comment]
> @farshidtz Do any of these fixes need to be backported?  Some of these sub-commands are going away in EdgeX 3.0 (Minnesota)

It should be sufficient and reasonable (since there is a workaround) to just fix them starting from 3.0 for any subcommand that continues to exist.
